src/ReplicatedStorage/Client/InstanceComponents/Misc/OceanDamage.lua

```diff
@@ -7,13 +7,16 @@ local OceanDamage = {}
 OceanDamage.__index = OceanDamage
 
 function OceanDamage.new(character: Model)
-	local humanoid = character:WaitForChild("Humanoid")
-
 	local self = setmetatable({}, OceanDamage)
 
+	self._humanoid = character:WaitForChild("Humanoid")
 	self._trove = Trove.new()
 	self._trove:Connect(RunService.Heartbeat, function(delta)
-		if character.HumanoidRootPart.Position.Y > OceanDamageConfig.Height then return end
+		local humanoid = self._humanoid
```

Comment:
Is this variable really necessary? It's already in self? Is there a performance reason or something I'm missing?